### PR TITLE
fix for vpn spinner spinning after connection

### PR DESCRIPTION
**Pull request by JEvansGit**
Hi, I believe this may fix this issue https://github.com/elementary/wingpanel-indicator-network/issues/93 . It seems to have fixed the issue on my laptop and desktop.

### CODE REVIEW

**Comment by jlnr** [comment]
I've tried installing this branch and I'm still seeing this issue.

File: src/common/Widgets/VpnMenuItem.vala [hunk 1]
```diff
@@ -93,6 +93,9 @@ public class Network.VpnMenuItem : Gtk.ListBoxRow {
                     critical ("An VPN is being connected but not active.");
                 }
                 break;
+            case State.CONNECTED_VPN:
+                hide_item (spinner);
```
Comment (jlnr) [inline src/common/Widgets/VpnMenuItem.vala]: `hide_item` is called at the start of the `update` method, so the spinner should already be hidden when control reaches this point.

Comment (JEvansGit) [reply src/common/Widgets/VpnMenuItem.vala]: I see. I think that the update method is being called too early as a result of the async operation ```    nm_client.activate_connection_async.begin``` being called in ```Widgets/VpnInterface.vala``` I am quite new to this so I am not too certain of a fix for this yet.

**Comment by JEvansGit** [comment]
Hi, I think that my first commit worked by chance due to the asynchronous ```nm_client.activate_connection_async.begin``` operation. I have reverted my change as it did not work most of the time. I have since added the ```check_vpn_connectivity``` function to check for VPN connectivity every half second for a maximum of 10 seconds. This seems to work on my computers.

File: src/common/Widgets/VpnMenuItem.vala [hunk 2]
```diff
@@ -110,4 +113,31 @@ public class Network.VpnMenuItem : Gtk.ListBoxRow {
...
+
+    /**
+    * Uses a timeout to check VPN connectivity
+    **/
+    private async void check_vpn_connectivity () {
+        if (!checking_vpn_connectivity) {
+
+            checking_vpn_connectivity = true;
+
+            for (int i = 0; i < 20; i++) {
```
Comment (JO3-W3B-D3V) [inline src/common/Widgets/VpnMenuItem.vala]: One rather subtle suggestion that I have, for future developers to make more sense of this, perhaps change the loop:

```for (int i = 0; i < 20; i++)``` 

I'd suggest replacing ```20``` with a more meaningful variable name. 
Perhaps make it a private static property on this class?

**Comment by marbetschar** [comment]
Function wise I can confirm this works. Tested on elementary OS 5.1.6 Hera. See the difference in behaviour in the below screencasts.

**1. Current stable version behaviour:**

Endless Spinner

**2. Applied vpn-spinner-fix branch behaviour:**

vpn-spinner-fix

**Pull request merged**